src/zino/statemodels.py

```diff
@@ -29,13 +29,31 @@ class InterfaceState(StrEnum):
     LOWER_LAYER_DOWN = "lowerLayerDown"
 
 
+class BFDSessState(Enum):
+    """The set of allowable BFD session states"""
+
+    ADMINDOWN = "adminDown"
+    DOWN = "down"
+    INIT = "init"
+    UP = "up"
+    NOSESSION = "noSession"
+
+
+class BFDState(BaseModel):
+    """Keeps BFD state for an interface"""
+
+    session_state: BFDSessState
+    session_index: Optional[int] = None
+
+
 class Port(BaseModel):
     """Keeps port state"""
 
     ifindex: int
     ifdescr: Optional[str] = None
     ifalias: Optional[str] = None
     state: Optional[InterfaceState] = None
+    bfd_state: Optional[BFDState] = None
```

Comment:
This will become a monster of a model, all things optional. The "if"-attributes are only seen on when type is portstate.